command/agent/retry_join.go

```diff
@@ -169,16 +169,16 @@ func (r *retryJoiner) RetryJoin() {
 
 	for {
 		var (
-			addrs []string
-			err   error
+			addrs, servers []string
+			err            error
 		)
 
 		for _, addr := range r.joinCfg.RetryJoin {
 
 			// If auto-discovery returns an error, log the error and
 			// fall-through, so we reach the retry logic and loop back around
 			// for another go.
-			servers, err := r.autoDiscover.Addrs(addr, r.logger)
+			servers, err = r.autoDiscover.Addrs(addr, r.logger)
```

Comment:
By not shadowing here I think we end up logging twice unless at least one `RetryJoin` returned an address. We'll log the error here and then again for the `join failed` warning. But maybe that's ok?

Reply:
Maybe? Or it can remain shadowed and the join failed log is only output when the `joinFunc` fails?

Reply:
I went back through the history and found this was just for surfacing the error if the `joinFunc` failed. So I updated it just for that.